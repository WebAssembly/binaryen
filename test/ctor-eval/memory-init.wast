(module
  (memory $0 1)
  (data (i32.const 0) "__________")
  (data (i32.const 20) "__________")
  (func "test1"
    ;; A store that can be evalled.
    (i32.store8
      (i32.const 4)
      (i32.const 100)
    )

    ;; A memory init cannot be evalled since ctor-eval flattens memory segments
    ;; atm. We lose the identity of them as a result as they are all merged
    ;; into a single big segment. We could fix that eventually.
    (memory.init 1
      (i32.const 0)
      (i32.const 0)
      (i32.const 1)
    )
  )
)

