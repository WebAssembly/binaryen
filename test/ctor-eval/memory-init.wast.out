(module
 (type $none_=>_none (func))
 (memory $0 1)
 (data (i32.const 0) "__________")
 (data (i32.const 20) "__________")
 (export "test1" (func $0))
 (func $0 (type $none_=>_none)
  (i32.store8
   (i32.const 4)
   (i32.const 100)
  )
  (memory.init 1
   (i32.const 0)
   (i32.const 0)
   (i32.const 1)
  )
 )
)
